### PR TITLE
Use latest php-cs-fixer 2.17.1

### PR DESCRIPTION
**Pull request by phil-davis**
The latest php-cs-fixer finds some new things. Code changes were needed in https://github.com/sabre-io/dav/pull/1316 for `sabre/dav`. We might as well consistently make sure to use at least this version 2.17.1 in all repos so that we are doing consistent code-style checking.

It complains about:
- underscore `_` in method names. 
- use of `die()` (wants it to be `exit()`)
- blank lines in switch-case

1) adjust code to make php-cs-fixer happy
2) Update windowszones timezone data to 2020-12-13
3) Fix various typos

### CODE REVIEW

File: lib/timezonedata/windowszones.php [hunk 1]
```diff
@@ -10,7 +10,7 @@
  * @license http://sabre.io/license/ Modified BSD License
  */
 
-return  [
+return [
```
Comment (phil-davis) [inline lib/timezonedata/windowszones.php:13]: This file says that it is auto-generated. But it has not been updated from the "source" any time recently.
php-cs-fixer did not like having 2 spaces here, so I made the edit anyway.
If someone re-generates the file from upstream some day, they might need to do a small edit like this to keep php-cs-fixer happy.

Comment (phil-davis) [reply lib/timezonedata/windowszones.php:13]: Oh, actually it is "auto-generated" from out own script. https://github.com/sabre-io/vobject/blob/master/bin/fetch_windows_zones.php

So it does not matter. I will update that script also...

**Comment by phil-davis** [comment]
@DeepDiver1975 @staabm please review. I do not feel comfortable merging it directly - there are quite a few lines of diff.

File: bin/fetch_windows_zones.php [hunk 2]
```diff
@@ -44,6 +43,6 @@
 
 echo "Formatting\n";
 
-exec(__DIR__.'/sabre-cs-fixer fix '.escapeshellarg($outputFile));
+exec(__DIR__.'/../vendor/bin/php-cs-fixer fix '.escapeshellarg($outputFile));
```
Comment (staabm) [inline bin/fetch_windows_zones.php:46]: hmm not sure its correct that this path changed? could you verify this bin still works?

**Review by staabm — APPROVED** [review]
except the one mentioned line, it looks good to me.

thanks!

File: bin/fetch_windows_zones.php (hunk 2 above)
Comment (phil-davis) [reply bin/fetch_windows_zones.php:46]: I ran `fetch_windows_zones.php` locally from the root of the repo in order to update the windows zones file.
That is how I found that the old `sabre-cs-fixer` is no longer, and made it work by doing this.

**Pull request merged**